src/Aspire.Hosting/Devcontainers/DevcontainerPortForwardingHelper.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+using System.Text.Json.Nodes;
+
+namespace Aspire.Hosting.Devcontainers;
+
+internal class DevcontainerPortForwardingHelper
+{
+    private const string CodespaceSettingsPath = "/home/vscode/.vscode-remote/data/Machine/settings.json";
+    private const string LocalDevcontainerSettingsPath = "/home/vscode/.vscode-server/data/Machine/settings.json";
+    private const string PortAttributesFieldName = "remote.portsAttributes";
+
+    public static async Task SetPortAttributesAsync(int port, string protocol, string label, CancellationToken cancellationToken = default)
```

Comment:
My one request is to add a semaphore slim here and make sure only one thing is calling this at a time. Added bonus if we use exclusive file locking but that's a bit trickier cross OS I think.